### ISSUE TITLE
Can Tevatron be used for Contriever fine-tuning ?

### ISSUE THREAD

**Issue by cramraj8** [texttron/tevatron#99]

**Comment by cramraj8** [comment]
@MXueguang  Thank you!

**Comment by cramraj8** [comment]
@MXueguang  does the new branch (main branch) currently incorporates both mean pooling and temperature in the loss?

**Comment by MXueguang** [comment]
--pooling mean \
--temperature xxx

yes. you can set these two argument during training to match the contriever setup.

**Comment by cramraj8** [comment]
Sounds good. It's already working great for me! I found from another thread that temperature is a sensitive parameter during contriever training. I also found with different batch size & temperature values, the performance substantially improves or drops. Any idea why and any optimum values you found? I am experimenting with multilingual datasets, TyDi and MIRACL.

**Comment by MXueguang** [comment]
@crystina-z can comments more here.

**Comment by cramraj8** [comment]
Is there any T5 encoder based Contriever pre-trained/ fine-tuned model available in HuggingFace? I would like to see the performance boost.

**Comment by crystina-z** [comment]
Hi @cramraj8 we actually didn't actively tune batch size and temperature. In the final config, we use batch=128 and the default temperature. You'll likely find a setting outperforming this, please open a PR if you do!
